src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,487 @@
+! stdlib_stringlist.f90 --
+!     Module for storing and manipulating lists of strings
+!     The strings may have arbitrary lengths, not necessarily the same
+!
+!     Note: very preliminary
+!
+!     TODO:
+!     insert( list_end, ... ) in an empty list?
+!
+!     Not implemented yet:
+!     insert a list or an array of character strings
+!     replace a string, list or an array of character strings
+!     concatenate a list with another list or an array
+!
+module stdlib_stringlist
+    implicit none
+
+    private
+    integer, parameter, public :: list_head = 0
+    integer, parameter, public :: list_end  = -1
+    public :: stringlist_type
+    public :: operator(//)
+
+
+    integer, parameter :: initial_size  = 20
+
+    type string_type
+        character(len=:), allocatable :: value
+    end type string_type
+
+    type stringlist_type
+        private
+        integer :: size = 0
+        type(string_type), dimension(:), allocatable :: string
+    contains
+        procedure :: destroy   => destroy_list
+        procedure :: insert    => insert_string
+        procedure :: get       => get_string
+        procedure :: length    => length_list
+        procedure :: sort      => sort_list
+        procedure :: index     => index_of_string
+        procedure :: index_sub => index_of_substring
+        procedure :: delete    => delete_strings
+        procedure :: range     => range_list
```

Comment:
First quick comment: shouldn't a `final` procedure (that potential calls the method `destroy`) be present?

Reply:
It could actually be a dummy - allocatables get cleaned up automatically.